packages/app-desktop/gui/note-viewer/index.html

```diff
@@ -388,6 +400,8 @@
 
 			contentElement.innerHTML = html;
 
+			rewriteFileUrls(event.options.mediaAccessKey);
+
```

Comment:
Does it run every time the viewer is refreshed? I wonder if that could be an issue in terms of performance?

Reply:
I'm now in the process of testing that!

Reply:
Test procedure:
1. Create a new note.
2. Attach a drawing.
3. Copy and paste the drawing's markup until there are 18,000 copies, each on a new line.
4. Disable all plugins.
5. Restart Joplin.
6. Switch to the performance tab and start recording.
7. Modify the note.

## 18,000 images

Based on this, it does have a performance impact:
![chart](https://github.com/user-attachments/assets/a894bfb4-22a3-4fd2-8fca-1c30ca392766)
In the above sample,  parse HTML is `383.42 ms` and `rewriteFileUrls` is `155.80 ms`.

Retrying with the same note, `rewriteFileUrls` has similar performance:
![chart](https://github.com/user-attachments/assets/70c1f29f-d7e5-4721-ad06-360face66e60)

## 100 images

With only 100 images, the `rewriteFileUrls` call doesn't seem to be visible in the performance graph:

![chart](https://github.com/user-attachments/assets/4d06cf6e-a3b5-49aa-b4a2-0e3c96a9ef51)